### ISSUE TITLE
MESSAGE_GET_VALUES Getter (and Setter?)

### ISSUE THREAD

**Issue by ausshir** [LiamBindle/PyVESC#3]
There is a need to read the MESSAGE_GET_VALUES messages as it gives all the RPM, Current, Power, Position etc. information.

The get-values branch contains the start of this work

As a first step, something like this should work:
```
def decode_test():
    buf_test = b'\x028\x04\x01\x14\x01\x14\x01\x14\x01\x14\x01\x14\x01\x14\x01\x14\x00\x00\x00\x18\x00\x00\x00\x06\x01\x10\x00\x00$u\x00\xef\x00\x00\x00\x0b\x00\x00\x00\x00\x00\x00\x01\t\x00\x00\x00\x00\x00\x00\xec\xbf\x00\x00\xec\xbf\x00\xa7g\x03'
    (a,b) = pyvesc.decode(buf_test)
    print(a, b)
```

Where buf_test represents the raw data coming off the serial port. This data is retreived by sending a blank message with id=4 (length=1), which also presents a challenge of having two message types with the same id.

A useful reference would be the VESC code in commands.c
https://github.com/vedderb/bldc/blob/master/commands.c#L162

**Comment by ausshir** [comment]
For no other reason than more test data, here's some more dumps from the VESC.

```
b'\x028\x04\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\x00\x00\x17\x00\x00\x00\x06\x01\x1d\x00\x00"\xc7
\x00\xee\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00X\x00\x00\x00\x00\x00\x00$%\x00\x00$%\x00\x0b\xea\x03'
b'\x028\x04\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\xf6\x00\x00\x00\x17\x00\x00\x00\x06\x01\x0c\x00\x00$\xed
\x00\xef\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00_\x00\x00\x00\x00\x00\x00)\xc4\x00\x00)\xc4\x00\xb3\xdc\x03'
b'\x028\x04\x00\xf7\x00\xf7\x00\xf7\x00\xf7\x00\xf6\x00\xf6\x00\xf6\x00\x00\x00\x17\x00\x00\x00\x06\x01\x0e\x00\x00%\n\x
00\xee\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00e\x00\x00\x00\x00\x00\x00/n\x00\x00/n\x00c^\x03'
b'\x028\x04\x00\xf7\x00\xf7\x00\xf7\x00\xf7\x00\xf7\x00\xf7\x00\xf7\x00\x00\x00\x17\x00\x00\x00\x06\x01\x08\x00\x00&\xef
\x00\xef\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00k\x00\x00\x00\x00\x00\x005\x07\x00\x005\x07\x00\x8e\x0e\x03'
b'\x028\x04\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\x00\x00\x17\x00\x00\x00\x06\x01\t\x00\x00&T\x00\
xed\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00r\x00\x00\x00\x00\x00\x00:\xa7\x00\x00:\xa7\x00)C\x03'
b'\x028\x04\x00\xf9\x00\xf9\x00\xf9\x00\xf9\x00\xf9\x00\xf9\x00\xf9\x00\x00\x00\x17\x00\x00\x00\x06\x01\x03\x00\x00+\xc3
\x00\xee\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00x\x00\x00\x00\x00\x00\x00@M\x00\x00@M\x00S\xca\x03'
b'\x028\x04\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\xf8\x00\x00\x00\x17\x00\x00\x00\x06\x01\x04\x00\x00&\xb6
\x00\xee\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x7f\x00\x00\x00\x00\x00\x00E\xea\x00\x00E\xea\x00\xf9\xbf\x03'
b'\x028\x04\x00\xfa\x00\xfa\x00\xfa\x00\xfa\x00\xfa\x00\xfa\x00\xfa\x00\x00\x00\x17\x00\x00\x00\x06\x01\x01\x00\x00(\xa7
\x00\xef\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x85\x00\x00\x00\x00\x00\x00K\x8e\x00\x00K\x8e\x00 \x94\x03'
```

**Comment by ausshir** [comment]
It turns out that the VESC's float16 and float32 datatypes are actually an integer with a scalar (so a decimal float in a way).

I added the ability in the last commit to specify an optional third argument that is a scalar. This is done using try/except blocks but there might be a more performant way?

Coupled with the new feature on the master branch, this issue is probably ready to close and pull into master. Needs some more testing on the physical hardware.

**Comment by ausshir** [comment]
I recently added an encoder mode setter, as well as an encoder position getter and setter.

Added a scalar option to setters as well, as the position setter was specified in milli-degrees in the VESC firmware.

Also, needed to change one of the unit tests, as id=21 conflicted with the duplicate id checking tests. How ironic.

Testing on physical hardware is complete and it all seems to work! :) Ready for a pull request.
